### PR TITLE
Issue #SB-27440 feat: Call notification service api call on each acti…

### PR DESCRIPTION
**Pull request by kamaleeswary**
…on from DF

### CODE REVIEW

**Review by VenkatKadiveti — APPROVED** [review]

File: helpers/constant.json [hunk 1]
```diff
@@ -31,10 +73,51 @@
...
-        "cdata": [ ],
+        "cdata": [],
+        "fromUid": "payload.user.uid",
+        "toUid": "payload.topic.uid",
+        "notificationObj": {
+            "headers": "payload.headers#",
+            "notificationData": {
+                "ids": "payload.userDetails.ids",
+                "createdBy": {
+                    "id": "payload.userDetails.id",
```
Comment (VenkatKadiveti) [inline helpers/constant.json:84]: createdBy object will have the logged in user, use createdBy object

File: helpers/notification.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  const result = mapfn.deepMap(refObject.notificationObj, (val) => {
+    let isExpression = (val.indexOf("?") > 0) ? true : false;
+    let isFunction = (val.indexOf("#") > 0) ? true : false;
+    let keyVal;
+    let value;
+    if (isExpression) {
+      let exprKey = val.substr(0, val.length - 1);
+      const objectVal = _.get(resp, String(exprKey));
+      if (typeof objectVal === 'object') {
+        value = true;
```
Comment (VenkatKadiveti) [inline helpers/notification.js:22]: why did you add true here

**Review by VenkatKadiveti — CHANGES_REQUESTED** [review]

**Review by VenkatKadiveti — APPROVED** [review]

File: proxy/proxyUtils.js [hunk 3]
```diff
@@ -58,7 +59,7 @@ const decorateRequestHeadersForPutApi = function () {
   }
 }
 
-const handleSessionExpiry = (proxyRes, proxyResData, req, res, error, data) => {
+const handleSessionExpiry = (proxyRes, proxyResData, req, res, error, data, triggerNotification = false) => {
```
Comment (vinukumar-vs) [inline proxy/proxyUtils.js:62]: You should not require the flag. You should check the constant object for the request URL has notificationObject, if it is present then trigger notification.

File: routes/nodebb.js [hunk 4]
```diff
@@ -12,6 +12,7 @@ const telemetryHelper = require('../helpers/telemetryHelper.js')
 const methodSlug = '/update';
 const nodebbServiceUrl = NODEBB_SERVICE_URL+ nodebb_api_slug;
 const _ = require('lodash')
+const triggerNotification = true;
```
Comment (vinukumar-vs) [inline routes/nodebb.js:15]: This is not required

Comment (vinukumar-vs) [reply routes/nodebb.js:15]: Instead of this, there should be env variable to trun-off notification for all API's
@VenkatKadiveti

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

**Pull request merged**